src/coreclr/inc/utsem.h

```diff
@@ -39,15 +42,20 @@ class UTSemReadWrite
     void UnlockRead();      // Unlock the object for reading
     void UnlockWrite();     // Unlock the object for writing
 
-#ifdef _DEBUG
+#if defined(_DEBUG) && defined(HOST_WINDOWS)
     BOOL Debug_IsLockedForRead();
     BOOL Debug_IsLockedForWrite();
-#endif //_DEBUG
+#endif // defined(_DEBUG) && defined(HOST_WINDOWS)
 
 private:
+#ifdef HOST_WINDOWS
     Volatile<ULONG> m_dwFlag;               // internal state, see implementation
     HANDLE          m_hReadWaiterSemaphore; // semaphore for awakening read waiters
     HANDLE          m_hWriteWaiterEvent;    // event for awakening write waiters
+#else // HOST_WINDOWS
+    bool m_initialized;
+    pthread_rwlock_t m_rwLock;
```

Comment:
How does performance of `pthread_rwlock_t` compares to the current implementation?

Reply:
Plugged `UTSemReadWrite` into corerun and some basic tests. The time cost for 1 million loops of single-thread lock-unlock under WSL:

PAL implementation: Read lock: 12350 us Write lock: 12379 us
pthreads implementation: Read lock: 11678 us Write lock: 15976 us

For reference, under Windows: Read lock: 13450 us Write lock: 12793 us